tests/providers/aws/e2e.py

```diff
@@ -226,6 +238,35 @@ def testStartVm(self):
     self.assertIn(volume_copy.volume_id,
                   [vol.volume_id for vol in instance.volumes.all()])
 
+  @typing.no_type_check
+  @IgnoreWarnings
+  def testCopyEBSSnapshotToS3(self):
+    """End to end test on AWS.
+
+    Test copying an EBS snapshot into S3.
+    """
+
+    if not self.s3_destination.startswith('s3://'):
+      self.s3_destination = 's3://' + self.s3_destination
+    path_components = s3.SplitStoragePath(self.s3_destination)
+
+    forensics.CopyEBSSnapshotToS3(
+      self.s3_destination,
+      self.snapshot_id,
+      'ebsCopy',
+      self.zone,
+      subnet_id=self.subnet_id,
+      security_group_id=self.security_group_id,
+      cleanup_iam=True)
+
+    aws_account = account.AWSAccount(self.dst_zone)
+    # pylint: disable=line-too-long
+    self.assertEqual(aws_account.s3.CheckForObject(path_components[0], '{0:s}/{1:s}/image.bin'.format(path_components[1], self.snapshot_id)), True)
```

Comment:
Please move the format string resulting from `path_components[1], self.snapshot_id` to its own variable and reuse that one throughout the calls.